vulnerabilities/api.py

```diff
@@ -128,13 +117,52 @@ def to_representation(self, instance):
         return data
 
     purl = serializers.CharField(source="package_url")
-    affected_by_vulnerabilities = VulnSerializerRefsAndSummary(
-        many=True, source="vulnerable_to", read_only=True
-    )
-    fixing_vulnerabilities = VulnSerializerRefsAndSummary(
-        many=True, source="resolved_to", read_only=True
-    )
-    fixed_packages = PackageSerializerFixedVulns(many=True, read_only=True)
+
+    affected_by_vulnerabilities = serializers.SerializerMethodField("get_affected_vulnerabilities")
+
+    fixing_vulnerabilities = serializers.SerializerMethodField("get_fixed_vulnerabilities")
+
+    def get_fixed_packages(self, package):
+        return Package.objects.filter(
+            name=package.name,
+            namespace=package.namespace,
+            type=package.type,
+            qualifiers=package.qualifiers,
+            subpath=package.subpath,
+            packagerelatedvulnerability__fix=True,
+        ).distinct()
+
+    def get_affected_vulnerabilities(self, package):
+        fixed_packages = self.get_fixed_packages(package=package)
+        qs = package.vulnerabilities.filter(packagerelatedvulnerability__fix=False)
```

Comment:
This is a bit confusing, `packagerelatedvulnerability__fix` is used with True in `get_fixed_packages` then overridden with False here?

Reply:
This qs is to get those vulnerabilities on the package that affects this that' why `packagerelatedvulnerability__fix` is set to False, whereas `get_fixed_packages` sends all the packages that matches name, namespace, type, qualifiers of that package and also fixes the vulnerability that's why it is set to True

Reply:
I get that, the confusing part is your approach of reusing the same base qs and overriding an existing lookup.